### PR TITLE
DiscoveryFeed: add hover state for tile

### PR DESCRIPTION
**Pull request by erikos**
This also makes the tile itself a bit bigger.

### CODE REVIEW

File: js/ui/discoveryFeedButton.js [hunk 1]
```diff
@@ -51,15 +51,25 @@ const DiscoveryFeedButton = new Lang.Class({
...
+        let iconFileHover = Gio.File.new_for_uri('resource:///org/gnome/shell/theme/discovery-feed-open-tab_hover.png');
+        this._giconHover = new Gio.FileIcon({ file: iconFileHover });
+        this._icon = new St.Icon({ gicon: this._giconNormal,
                                    style_class: 'discovery-feed-icon',
                                    track_hover: true });
         this.parent({ name: 'discovery-feed',
                       child: this._icon,
                       x_align: Clutter.ActorAlign.CENTER,
                       y_align: Clutter.ActorAlign.CENTER });
+        this.connect('notify::hover', Lang.bind(this, this._onHoverChanged));
```
Comment (smspillaz) [inline js/ui/discoveryFeedButton.js]: Hmm, my preference here would be to use the css `background-image` attribute with a `:hover` state.

Comment (erikos) [inline js/ui/discoveryFeedButton.js]: Thanks Sam for pushing me to this, that looks much better now indeed. I was not sure about the exact alignment in css, there are different styles used, I followed the page-indicator example.

**Comment by erikos** [comment]
I sneaked in here a small fix for CodeView to not have another PR: Gjs-Message: JS WARNING: [/home/mario/work/endless/shell-rebase/gnome-shell/js/ui/codeView.js 1006]: anonymous function does not always return a value

File: js/ui/discoveryFeedButton.js [hunk 2]
```diff
@@ -51,12 +51,10 @@ const DiscoveryFeedButton = new Lang.Class({
     Extends: St.Button,
 
     _init: function() {
-        let iconFile = Gio.File.new_for_uri('resource:///org/gnome/shell/theme/discovery-feed-open-tab.png');
-        let gicon = new Gio.FileIcon({ file: iconFile });
-        this._icon = new St.Icon({ gicon: gicon,
-                                   style_class: 'discovery-feed-icon',
-                                   track_hover: true });
+        this._icon = new St.Icon({style_class: 'discovery-feed-tile-icon',
```
Comment (cosimoc) [inline js/ui/discoveryFeedButton.js]: If you set the background with CSS, you should not need this `St.Icon` child on the button anymore. Try simply setting the background image on the `.discovery-feed-tile` style class of the button (which however I would rename to `.discovery-feed-button`)

Comment (erikos) [inline js/ui/discoveryFeedButton.js]: Event better, thanks for the suggestion - it really starts to look nice clean code now.

**Comment by cosimoc** [comment]
@erikos Can you rebase this to the latest master? Somehow it has a commit of mine that's already in master.

**Comment by erikos** [comment]
@cosimoc, rebased, sorry must have happened in rebasing here when tweaking.

**Pull request merged**